fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -1239,6 +1246,15 @@ class ApplyMetadataMutationsImpl {
 
 } // anonymous namespace
 
+void ResolverData::initGroupMessageBuilders(Version commitVersion) {
+	if (!SERVER_KNOBS->PROXY_USE_RESOLVER_PRIVATE_MUTATIONS)
+		return;
+
+	const auto& groups = tLogGroupCollection->groups();
+	ASSERT_WE_THINK(toCommit && toCommit->pGroupMessageBuilders->empty());
+	toCommit->addTLogGroups(groups, commitVersion);
```

Comment:
do we send message to all tlog groups as long as there is any metadata mutation in the commit? I thought we only do this when the shard boundary changes.

Reply:
This is to initialize all groups, because we don't know which group will have private mutations. Resolver calls `toCommit.getGroupMutations(writtenGroups)` later to only extract private mutations for the `writtenGroups`, not all groups.

Reply:
Can you add a comment what is initialized, and why it is needed here?

Reply:
Done.